src/NUnitFramework/framework/Internal/Results/TestCaseResult.cs

```diff
@@ -87,9 +88,9 @@ public override bool HasChildren
         /// <summary>
         /// Gets the collection of child results.
         /// </summary>
-        public override IList<ITestResult> Children
+        public override ConcurrentQueue<ITestResult> Children
         {
-            get { return new ITestResult[0]; }
+            get { return EmptyChildQueue; }
```

Comment:
In the original code, I considered a single instance for efficiency but decided to avoid it because this member of a TestCaseResult should never be accessed. In fact, for a while, I left it null to see if any error arose. None did. It may be acceptable to throw a NotImplementedException here.

Alternatively, if we put ITestResult.Children back to being some simple interface, we could return a null implementation of it. In that case, it wouldn't matter whether we used a single instance or a new instance.

I did try working with two different interfaces but the code is really much better if every test returns an ITestResult.